### PR TITLE
Jerk limits

### PR DESCRIPTION
**Pull request by duncancmt-dexai**

### CODE REVIEW

File: toppra/dracula/three_segment_spline.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+                if scale is None or scale < joint_scale:
+                    scale = joint_scale
+
+        if scale is None or scale < 1:
+            scale = 1
+        scale = scale ** (1/3)
+
+        C_ret[i,::,2] /= scale
+        C_ret[i,::,1] /= scale**2
+        C_ret[i,::,0] /= scale**3
```
Comment (duncancmt-dexai) [inline toppra/dracula/three_segment_spline.py]: I couldn't wrap my head around the correct algebra for scaling the coefficients. I'll come back at this tomorrow

**Comment by duncancmt-dexai** [comment]
This PR (and DexaiRobotics/dracula#648) will be closed/deleted on Friday, August 2nd because it will be superseded by other work

**Comment by tarquasso** [comment]
@duncancmt-dexai How much work would it take to finish this PR?

**Comment by tarquasso** [comment]
please leave this PR open until the solution from Gavin comes in.

**Comment by duncancmt-dexai** [comment]
It shouldn't require much extra work to get this merged. I'll do that immediately

**Comment by tarquasso** [comment]
is this ready for review?

**Comment by duncancmt-dexai** [comment]
Ready for review, but still needs on-robot testing

**Comment by tarquasso** [comment]
@duncancmt-dexai Please make two RunTopp calls: one with jerk limit at start and end, and one with jerk limits for all segments.

File: toppra/dracula/topp_dracula_interface.py [hunk 2]
```diff
@@ -11,11 +11,11 @@
 
 logging.basicConfig()
 
-from .zero_acceleration_start_end import ZeroAccelerationAtStartAndEnd
+from .three_segment_spline import ThreeSegmentRespline
 
 _CALLS_TO_RUN_TOPP = 0
 
-def RunTopp(knots_ext, vlim, alim,robot_command_rate, return_spline_parameters = False, topp_breaks_count = 1001, debug_active = False):
+def RunTopp(knots_ext, vlim, alim, jlim, robot_command_rate, return_spline_parameters = False, topp_breaks_count = 1001, debug_active = False):
```
Comment (tarquasso) [inline toppra/dracula/topp_dracula_interface.py:18]: create a seaprate RunTopp version for the jerk limitation on all segments

File: toppra/dracula/topp_dracula_interface.py [hunk 3]
```diff
@@ -79,7 +79,7 @@ def RunTopp(knots_ext, vlim, alim,robot_command_rate, return_spline_parameters =
     if debug_active:
     	csplcp= copy.deepcopy(jnt_traj.cspl)
 
-    ZeroAccelerationAtStartAndEnd(jnt_traj.cspl)
```
Comment (tarquasso) [inline toppra/dracula/topp_dracula_interface.py:82]: keep this alive in one of the versions

**Review by tarquasso — CHANGES_REQUESTED** [review]
Please make two RunTopp calls: old version that only fixes start and end, and this new version you are proposing here.

**Comment by duanyutong** [comment]
closing as this appears no longer needed

**Pull request closed without merging**